packages/run-protocol/test/amm/vpool-xyk-amm/test-amm-calc.js

```diff
@@ -0,0 +1,83 @@
+// @ts-check
+import { test } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
+import { AmountMath as m, makeIssuerKit } from '@agoric/ertp';
+import { natSafeMath } from '@agoric/zoe/src/contractSupport/index.js';
+import fc from 'fast-check';
+import {
+  imbalancedRequest,
+  balancesToReachRatio,
+} from '../../../src/vpool-xyk-amm/addLiquidity.js';
+
+const { add, multiply } = natSafeMath;
+
+const { brand: brandX } = makeIssuerKit('X');
+const { brand: brandY } = makeIssuerKit('Y');
+
+const oneB = 1_000_000_000n;
+// const dec18 = 1_000_000_000_000_000_000n; // 18 decimals as used in ETH
```

Comment:
Try taking out `oneB` and all the `max: oneB` stuff, as well as this dec18 comment?

It was prompted by failing tests. The original thought was that `oneB * dec18` would represent a practical limit: 1B ETH, with 10^18 wei to the ETH. Then that still failed, so I went down to just 1B. Given 10^6 urun to the RUN, that only covers up to 10^3 RUN, which is clearly not enough to cover all intended use of the AMM.

Now that the implementation covers the edge cases, we shouldn't need this limit.

Not critical, I guess.

Reply:
We had tried an earlier round of property-based testing, and ended up removing it because it was too expensive to include in tests. Is your expectation that this would be checked in?

Reply:
I have a mild preference for having this checked in.

There is now precedent in ERTP for prroperty-based testing. For straightforward calculation functions like this, I don't notice much cost. But that doesn't mean it's not there...